### PR TITLE
Veteran information page

### PR DESCRIPTION
**Pull request by akainic**
The first page in the veteran information chapter. Built from wireframes here: https://app.moqups.com/greg@adhocteam.us/l32hJ36b/view/page/a3c1b5b8f

Need feedback on a couple of things in particular:
1. @jbalboni I changed the `ensureFieldsInitialized` reducer back to what we had for HCA because I was having trouble getting the education version working (due to nested veteran object). But happy to optimize it once we can make sure it works.
2. @cweber are there too many questions on this page? You need to scroll before you see the Continue button, and something we tried to do on HCA is keep each section as short as possible. We can always figure this out later, just something to keep in mind.

personalinformationpage

### CODE REVIEW

File: src/js/edu-benefits/reducers/veteran/index.js [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+import { makeField, dirtyAllFields } from '../../../common/model/fields';
 
 const blankVeteran = {
   benefitsRelinquished: makeField(''),
   chapter30: false,
   chapter1606: false,
   chapter32: false,
-  chapter33: false
+  chapter33: false,
+  veteranFullName: {
```
Comment (jbalboni) [inline src/js/edu-benefits/reducers/veteran/index.js]: I've been trying to match these field names to the json schema: https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/dist/edu-benefits-schema.json

File: src/js/edu-benefits/reducers/veteran/index.js [hunk 2]
```diff
@@ -17,11 +30,21 @@ export default function veteran(state = blankVeteran, action) {
...
-      return action.fields.reduce((vet, field) => {
-        return _.isObject(vet[field]) ?
-          _.set([field, 'dirty'], true, vet) :
-          vet;
-      }, state);
+      const newState = Object.assign({}, state);
+
+      if (action.parentNode) {
+        action.fields.map((field) => {
+          Object.assign(newState[action.parentNode][0][field], dirtyAllFields(newState[action.parentNode][0][field]));
```
Comment (jbalboni) [inline src/js/edu-benefits/reducers/veteran/index.js]: I have a change for this in my page, too. One thing that seems weird about the way it is in HCA is that it's always updating the first record of the parentNode array. It seemed like this was causing problems with the GrowableTable I was using. 

In any case, I think this PR will be merged before mine, so I'll make sure whatever change I make works with this page and ping you to look at the change.

File: src/js/edu-benefits/utils/validations.js [hunk 3]
```diff
@@ -387,18 +389,20 @@ function isValidServiceInformation(data) {
       isNotBlank(data.dischargeType.value);
 }
 
-function isBenefitsInformationSectionValid(data) {
+function isValidBenefitsInformationSection(data) {
```
Comment (jbalboni) [inline src/js/edu-benefits/utils/validations.js]: Ah, good catch.

**Comment by jbalboni** [comment]
screen shot 2016-09-01 at 2 16 02 pm
screen shot 2016-09-01 at 2 16 13 pm

**Comment by jbalboni** [comment]
@akainic LGTM. The one real question I had is about matching field names to the json schema.

File: src/js/edu-benefits/utils/validations.js (hunk 3 above)
Comment (akainic) [inline src/js/edu-benefits/utils/validations.js]: It doesn't really matter either way just wanted it to be consistent :)

File: src/js/edu-benefits/reducers/veteran/index.js (hunk 1 above)
Comment (akainic) [inline src/js/edu-benefits/reducers/veteran/index.js]: Good point. I was trying to make this consistent with the names HCA used. Perhaps the schema for education should change then so it's all consistent?

Comment (jbalboni) [inline src/js/edu-benefits/reducers/veteran/index.js]: I think that makes sense, assuming we like the names in hca (I haven't looked at closely).

Comment (akainic) [inline src/js/edu-benefits/reducers/veteran/index.js]: Created this ticket in the JSON schema repo for discussion: https://github.com/department-of-veterans-affairs/vets-json-schema/issues/11

**Pull request merged**